### PR TITLE
Add an illustrative example to ?GForce when sorting locale matters

### LINKED ISSUE #5331: Unexpected result for max of character variable by group
I was surprised by this:

```
DT <- data.table(group = c("g1", "g1", "g2", "g2"),
                 x = c("alice", "Bob", "carol", "david"))

DT
#    group     x
# 1:    g1 alice
# 2:    g1   Bob
# 3:    g2 carol
# 4:    g2 david

DT2 <- DT[, .(m1 = max(x)), by = "group"]

DT2
#    group    m1
# 1:    g1 alice
# 2:    g2 david

DT3 <- DT[, .(m1 = max(x), m2 = max(tolower(x))), by = "group"]

DT3
#    group    m1    m2
# 1:    g1   Bob   bob
# 2:    g2 david david

DT[group == "g1", max(x)]
# [1] "Bob"

sessionInfo()

R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.14.2

loaded via a namespace (and not attached):
[1] compiler_4.1.2
```

Why are `DT2$m1` and `DT3$m1` different?

And why is `DT2[group == "g1", m1]` not the same as `DT[group == "g1", max(x)]`?

Thanks.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #5331

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -609,5 +609,7 @@
...
 
+17. Some clarity is added to `?GForce` for the case when subtle changes to `j` produce different results because of differences in locale. Because `data.table` _always_ uses the "C" locale, small changes to queries which activate/deactivate GForce might cause confusingly different results when sorting is involved, [#5331](https://github.com/Rdatatable/data.table/issues/5331). The inspirational example compared `DT[, .(max(a), max(b)), by=grp]` and `DT[, .(max(a), max(tolower(b))), by=grp]` -- in the latter case, GForce is deactivated owing to the _ad-hoc_ column, so the result for `max(a)` might differ for the two queries. An example is added to `?GForce`. As always, there are several options to guarantee consistency, for example (1) use namespace qualification to deactivate GForce: `DT[, .(base::max(a), base::max(b)), by=grp]` or (2) turn off GForce with `options(datatable.optimize = 0)`. Thanks @markseeto for the example and @michaelchirico for the improved documentation.
```
Comment (tdhock) [inline NEWS.md:612]: this is a great news item

Comment (jangorecki) [reply NEWS.md:612]: Another option worth to mention is setting up locale to C

**Comment by MichaelChirico** [comment]
@TysonStanley are we in the thick of the release process already? Wondering if maybe this doc-only change should be shipped with 1.15.0.

It's not urgent though, so if you've already started packaging up 1.15.0 for CRAN, no need to bother -- we'll just merge to 1-15-99.

**Comment by TysonStanley** [comment]
Was waiting on a fix from @ben-schwen and @tdhock added the change in the description too so this would be just fine.

**Comment by MichaelChirico** [comment]
> Was waiting on a fix from @ben-schwen

Is that different from #5887? That was merged / results in revdep show nothing new. Sorry but not sure I follow the second half of your comment.

**Comment by TysonStanley** [comment]
Great, then I would say to include this and then tomorrow I'll restart the release process.

As for the second part, Toby changed ctr to aut for you, Toby, and Jan as well but I had seen that was merged, just missed the one from Ben.

**Comment by MichaelChirico** [comment]
OK, rebased to `master` again. Ready for merge after final review.

**Comment by MichaelChirico** [comment]
> As for the second part, Toby changed ctr to aut for you, Toby, and Jan as well but I had seen that was merged, just missed the one from Ben.

NB: Ben is not currently a committer: https://github.com/orgs/Rdatatable/teams/maintainers

**Comment by TysonStanley** [comment]
> > As for the second part, Toby changed ctr to aut for you, Toby, and Jan as well but I had seen that was merged, just missed the one from Ben.
> 
> NB: Ben is not currently a committer: https://github.com/orgs/Rdatatable/teams/maintainers

Right, I meant Ben's work on #5887 . Only you, Jan, and Toby are committers.

**Review by ben-schwen — COMMENTED** [review]
GForce is already turned off when `options(datatable.optimize=1L)`. Might want to adjust that or say that optimize=0 turns optimizations off

**Review by jangorecki — APPROVED** [review]

**Pull request merged**